processing/manual_correction.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Run this script in results/data folder
+#
+# USAGE:
+# Define files.yaml file as following:
+#   ---
+#   FILES_SEG:
+#   - sub-amu01_T1w_RPI_r.nii.gz
+#   - sub-amu01_T2w_RPI_r.nii.gz
+#   - sub-brnoCeitec01_T2star_rms.nii.gz
+#   FILES_LABEL:
+#   - sub-amu01
+#   - sub-brnoUhb01
+#
+#
+# THEN RUN:
+#   PATH_TO_SPINEGENERIC/processing/manual_correction.sh files.yaml
+#
+# Authors: Julien Cohen-Adad, Jan Valosek
+
+
+# function for parsing input yaml file
+yaml_parser() {
+
+  INPUT_YAML=$1
+
+  # tr: delete all newlines begining with dash (-), so convert input file into one line without dashes
+  # (this also deletes all other dashes, so it is necessary to use last sed)
+  # 1st sed: delete "FILES_LABEL:" string and everything after it
+  # 2nd sed: delete "FILES_SEG:" string itself
+  # 3rd sed: replace "sub" by "sub-" (tr command deleted all dashes (-))
+  FILES_SEG=$(cat $INPUT_YAML | tr -d '\n-' | sed 's/FILES_LABEL:.*//' | sed 's/FILES_SEG://' | sed 's/sub/sub-/g')
+  # tr: delete all newlines begining with dash (-), so convert input file into one line without dashes
+  # (this also deletes all other dashes, so it is necessary to use last sed)
+  # 1st sed: delete everything before "FILES_LABEL:" string including string "FILES_LABEL:" itself
+  # 2nd sed: replace "sub" by "sub-" (tr command deleted all dashes (-))
+  FILES_LABEL=$(cat $INPUT_YAML | tr -d '\n-' | sed 's/.*FILES_LABEL://' | sed 's/sub/sub-/g')
+}
+
+# Print help, if invalid input
+if [[ $# != 1 ]] || [[ $1 == "--help" ]] || [[ $1 == "-h" ]];then
+
+  echo -e "Invalid input. \n\nPlease define files.yaml file with list of segmentations to correct"
+  echo -e "and list of subjects to correct labeling on (one file perline)."
+  echo -e "\nExample:"
+  echo -e "\t---"
+  echo -e "\tFILES_SEG:"
+  echo -e "\t- sub-amu01_T1w_RPI_r.nii.gz"
+  echo -e "\t- sub-amu01_T2w_RPI_r.nii.gz"
+  echo -e "\t- sub-brnoCeitec01_T2star_rms.nii.gz"
+  echo -e "\tFILES_LABEL:"
+  echo -e "\t- sub-amu01"
+  echo -e "\t- sub-brnoUhb01"
+  echo -e "\nThen run this script in results/data folder:\n\tPATH_TO_SPINEGENERIC/processing/manual_correction.sh files.yaml"
+
+else
+
+  # call yaml_parser function
+  yaml_parser $1
+
+  # Folder to output the manual segmentations and labels
+  PATH_SEGMANUAL="../../seg_manual"
```

Comment:
strong assumption-- this should at least be documented somewhere-- or could be input arg

Reply:
You are right 👍 I have mentioned it in the script description. But sure, it could be input arg. We could you e.g. `getopts` built-in bash function ([example](https://sookocheff.com/post/bash/parsing-bash-script-arguments-with-shopts/)).

Reply:
actually, why not making it a python script? advantages:
- can be added as entry point script in the setup.py so could be called from anywhere
- python is friendlier (better argparse, better yaml parser, easier debugging, etc.)
- more harmonious for the package (we already have a few python scripts)

Reply:
Good idea. Many of mentioned problems will be easier to solve. I will start with building of it. I will create new PR when draft will be ready.